internal/dataplane/kong_client.go

```diff
@@ -134,6 +134,14 @@ type KongClient struct {
 
 	// clientsProvider allows retrieving the most recent set of clients.
 	clientsProvider AdminAPIClientsProvider
+
+	// hasTranslationErrorChan is used to notify konnect node agent the whether
+	//	error happened in traslating k8s objects to kong configuration.
+	hasTranslationErrorChan chan bool
+
+	// sendConfigErrorChan is used to notify whether error happened in sending
+	// translated configurations to kong.
+	sendConfigErrorChan chan error
```

Comment:
Could we have a separate type that will handle this and inject it into `KongClient`?  That could be named `ConfigStatusNotifier` and have a method `Notify(ConfigStatus)`. In case we don't have Konnect sync enabled, we could use noop implementation so that `KongClient` wouldn't have to include logic for checking if the channels exist or not.

Recently we were trying to move things out from the `KongClient` if possible as it tends to become a "god object" with too many responsibilities.

Reply:
moved to `ConfigStatusNotifier` interface.